engine/src/main/java/org/hibernate/validator/internal/engine/ValidatorImpl.java

```diff
@@ -463,15 +463,19 @@ private void validateConstraintsForCurrentGroup(BaseBeanValidationContext<?> val
 			// if the current class redefined the default group sequence, this sequence has to be applied to all the class hierarchy.
 			if ( defaultGroupSequenceIsRedefined ) {
 				Iterator<Sequence> defaultGroupSequence = hostingBeanMetaData.getDefaultValidationSequence( valueContext.getCurrentBean() );
-				Set<MetaConstraint<?>> metaConstraints = hostingBeanMetaData.getMetaConstraints();
+				Set<MetaConstraint<?>> classMetaConstraints = hostingBeanMetaData.getClassMetaConstraints();
+				Set<MetaConstraint<?>> propertyMetaConstraints = hostingBeanMetaData.getPropertyMetaConstraints();
 
 				while ( defaultGroupSequence.hasNext() ) {
 					for ( GroupWithInheritance groupOfGroups : defaultGroupSequence.next() ) {
 						boolean validationSuccessful = true;
 
 						for ( Group defaultSequenceMember : groupOfGroups ) {
-							validationSuccessful = validateConstraintsForSingleDefaultGroupElement( validationContext, valueContext, validatedInterfaces, clazz,
-									metaConstraints, defaultSequenceMember ) && validationSuccessful;
+							boolean propertyValidationResult = validateConstraintsForSingleDefaultGroupElement( validationContext, valueContext, validatedInterfaces, clazz, propertyMetaConstraints, defaultSequenceMember );
+							validationSuccessful = propertyValidationResult && validationSuccessful;
+							if ( !validationContext.isIgnoreClassLevelOnPropertyFailModeEnabled() || propertyValidationResult ) {
```

Comment:
I'm wondering if having these two calls could slow down things. I wonder if we should have a shortcut if this mode is disabled.

Also wondering if we should test if the set is empty (can't say if it will be a win though but it looks like something we should do - and bench).

In any case, we should test it.